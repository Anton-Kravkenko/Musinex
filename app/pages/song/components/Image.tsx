import { AnimatedImage } from '@/animation/global'
import { songAnimationValue } from '@/pages/song/animation/songAnimationTypes'
import { useSongAnimation } from '@/pages/song/animation/useSongAnimation'
import { WindowHeight, WindowWidth } from '@/utils/screen'
import { FC } from 'react'
import { View } from 'react-native'
import { Track } from 'react-native-track-player'

interface IImage  extends songAnimationValue, Pick<Track, 'artwork'>{}

const Image:FC<IImage> = ({artwork, isOpen}) => {
	const {ImageAnimation} = useSongAnimation(isOpen)
	return <View
		style={{ marginTop: WindowHeight * 0.06}}
		className='relative items-center  justify-center self-center'
	>
		<AnimatedImage
			source={{
				uri: String(artwork),
				height: WindowWidth * 0.8,
				width: WindowWidth * 0.8,
			}}
			style={[{
				borderRadius: 15,
			}, ImageAnimation]}
			
			resizeMode={'cover'}
			className=' relative items-center  justify-center'
		/>
	</View>
}

export default Image