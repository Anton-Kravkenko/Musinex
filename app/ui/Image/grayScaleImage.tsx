import React, { FC } from 'react'
import { Image, View } from 'react-native'
import { IImage } from './IImage'

const GrayScaleImage:FC<IImage> = (props,{...rest}) => {
	
	return (
				<View>
						<Image
							source={{
					uri: props.source,
						width: props.width,
						height: props.height,
				}}
				style={{
					width: props.width,
						height: props.height,
						resizeMode: "cover",
						tintColor: "#333",
				}} className={props.classNames}
				/>
				<Image
					source={{
						uri: props.source,
						width: props.width,
						height: props.height,
					}}
					style={{
						position: "absolute",
						opacity:0.4,
						width: props.width,
						height: props.height,
						resizeMode: "cover",
					}}
					{...rest}
					className={props.classNames}
				/>
				</View>
	)
}

export default GrayScaleImage
