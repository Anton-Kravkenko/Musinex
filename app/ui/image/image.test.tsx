import Image from '@/ui/image/image'
import { cleanup, render } from '@testing-library/react-native'

afterEach(() => cleanup())
describe('Image', () => {
	test('Render', () => {
		const imageRender = render(
			<Image
				width={20}
				height={20}
				borderRadius={10}
				url={
					'https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png'
				}
				style={{
					width: 20
				}}
				className='mb-5'
			/>
		)
		expect(imageRender).toMatchSnapshot()
	})
})
