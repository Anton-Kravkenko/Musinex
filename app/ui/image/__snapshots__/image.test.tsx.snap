// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Render 1`] = `
<View>
  <Image
    className="mb-5"
    onLoadEnd={[Function]}
    source={
      {
        "height": 20,
        "uri": "https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png",
        "width": 20,
      }
    }
    style={
      [
        {
          "borderRadius": 10,
          "display": "none",
          "height": 20,
          "width": 20,
        },
        {
          "width": 20,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#888888",
          "borderRadius": 6,
          "height": 20,
          "width": 20,
        },
        [
          {
            "borderRadius": 10,
          },
          {
            "width": 20,
          },
        ],
      ]
    }
    testID="skeleton"
  />
</View>
`;
